sycl/source/kernel_bundle.cpp

```diff
@@ -112,17 +112,18 @@ bool kernel_bundle_plain::is_specialization_constant_set(
 //////////////////////////////////
 ///// sycl::detail free functions
 //////////////////////////////////
+inline auto getDeviceComparisonLambda() {
+  return [](device a, device b) { return a.impl < b.impl; };
+}
```

Comment:
Ah, I see that we already have a friend declaration for that in device.hpp, that's not currently defined anywhere. Do you think it would make sense to get rid of it and just work with getSyclObjImpl instead just to keep the number of friend functions to a minimum?

Reply:
@KseniyaTikhomirova Right, I noticed that too late, see my second comment.

Reply:
I suppose that having friend function here may be better just because this comparison function will be invoked many times and getSyclObjImpl in this case too,  instead of direct access to impl pointer.

Reply:
no need to split, give me a few minutes to update